app/src/main/java/com/iic/shopingo/ui/trip_flow/activities/ManageTripActivity.java

```diff
@@ -154,15 +145,25 @@ public void onRequestSelected(IncomingRequest request) {
 
   @Override
   public void onDiscardTripDialogOK() {
-    // TODO: Invoke API call to end current trip
-    if (backPressed) {
-      super.onBackPressed();
-    } else {
-      NavUtils.navigateUpFromSameTask(this);
-    }
-
-    backPressed = false;
-    upPressed = false;
+    ApiTask<ApiResult> task = new ApiTask<>(getSupportFragmentManager(), "Ending trip...", new EndTrip(CurrentUser.getToken()));
+    task.execute().continueWith(new Continuation<ApiResult, Object>() {
+      @Override
+      public Object then(Task<ApiResult> task) throws Exception {
+        if (!task.isFaulted() && !task.isCancelled()) {
+          if (backPressed) {
+            ManageTripActivity.super.onBackPressed();
+          } else {
+            NavUtils.navigateUpFromSameTask(ManageTripActivity.this);
+          }
+
+          backPressed = false;
+          upPressed = false;
```

Comment:
come to think of it, we just need the `backPressed` boolean, no need for them both.